cli/tests/test_git_colocated.rs

```diff
@@ -975,3 +975,354 @@ fn stopgap_workspace_colocate(
+
+        let second_git = git2::Repository::open(&second_path).unwrap();
+        assert!(second_git.head_detached().unwrap());
+        let second_head = second_git.head().unwrap();
+        assert_eq!(
+            second_head.peel_to_commit().unwrap().id().to_string(),
+            followup_commit,
+            "second workspace's git HEAD should have advanced to {followup_commit}"
+        );
+    }
+
+    // Finally, test imports. Test that a commit written to HEAD in one workspace
+    // does not get imported by the other workspace.
+
+    // Write in default, expect second not to import it
+    let new_commit = test_independent_import(&test_env, &repo_path, &second_path, &followup_commit);
+    // Write in second, expect default not to import it
+    test_independent_import(&test_env, &second_path, &repo_path, &new_commit);
+
+    fn test_independent_import(
+        test_env: &TestEnvironment,
+        commit_in: &Path,
+        no_import_in_workspace: &Path,
+        workspace_at: &str,
+    ) -> String {
+        // Commit in one workspace
+        let mut repo = gix::open(commit_in).unwrap();
+        {
+            use gix::config::tree::*;
+            let mut config = repo.config_snapshot_mut();
+            let (name, email) = ("JJ test", "jj@example.com");
+            config.set_value(&Author::NAME, name).unwrap();
+            config.set_value(&Author::EMAIL, email).unwrap();
+            config.set_value(&Committer::NAME, name).unwrap();
+            config.set_value(&Committer::EMAIL, email).unwrap();
+        }
+        let tree = repo.head_tree_id().unwrap();
+        let current = repo.head_commit().unwrap().id;
+        let new_commit = repo
+            .commit(
+                "HEAD",
+                format!("empty commit in {}", commit_in.display()),
+                tree,
+                [current],
+            )
+            .unwrap()
+            .to_string();
+
+        let (check_git_head, stderr) = test_env.jj_cmd_ok(
+            no_import_in_workspace,
+            &["log", "--no-graph", "-r", "git_head()", "-T", "commit_id"],
+        );
+        // Asserting stderr is empty => no import occurred
+        assert_eq!(
+            stderr,
+            "",
+            "Should not have imported HEAD in workspace {}",
+            no_import_in_workspace.display()
+        );
+        // And the commit_id should be pointing to what it was before
+        assert_eq!(
+            check_git_head,
+            workspace_at,
+            "should still be at {workspace_at} in workspace {}",
+            no_import_in_workspace.display()
+        );
+
+        // Now we import the new HEAD in the commit_in workspace, so it's up to date.
+        let (check_git_head, stderr) = test_env.jj_cmd_ok(
+            commit_in,
+            &["log", "--no-graph", "-r", "git_head()", "-T", "commit_id"],
+        );
+        assert_eq!(
+            stderr,
+            "Reset the working copy parent to the new Git HEAD.\n",
```

Comment:
nit: better to use `insta::allow_duplicates!`? It's tedious to update the expectation on wording changes.